src/pages/Mainpage/index.tsx

```diff
@@ -0,0 +1,143 @@
+import { useEffect, useRef, useState } from "react";
+import moment from "moment";
+import { useRecoilState, useRecoilValue } from "recoil";
+import { useParams } from "react-router-dom";
+
+import ChatBubble from "@/components/ChatBubble";
+import { IChatData } from "@/interfaces";
+import { chatToggleState, talkState } from "@/states";
+import DummyChatData from "@/assets/datas/chatData.json";
+import alienSrc from "@/assets/images/alien.png";
+import { ChatLineWrapper, ChatTime, MainpageContainer } from "./index.element";
+
+const Mainpage = () => {
+  const [chatData, setChatData] = useState(DummyChatData);
+  const talker = useRecoilValue(talkState);
+  const mainpageRef = useRef<HTMLDivElement>(null);
+  const [chatRefresh, setChatRefresh] = useRecoilState(chatToggleState);
+  const { chatId } = useParams();
+  const [chatLength, setChatLength] = useState(0);
+
+  const setLocalStorage = () => {
+    if (localStorage.getItem(`chatData_${chatId}`) === null) {
+      localStorage.setItem(
+        `chatData_${chatId}`,
+        JSON.stringify(
+          chatData.find((el) => el.id === Number(chatId))?.chatData
+        )
+      );
+    } else {
+      localStorage.removeItem(`chatData_${chatId}`);
+      localStorage.setItem(
+        `chatData_${chatId}`,
+        JSON.stringify(
+          chatData.find((el) => el.id === Number(chatId))?.chatData
+        )
+      );
+    }
+  };
+
+  const handleLike = (id: string) => {
+    let target = chatData
+      .find((el) => el.id === Number(chatId))
+      ?.chatData.find((el) => el.createdAt === id)!;
+    target.like = true;
+    setChatData(
+      chatData.map((el) => {
+        if (el.id === Number(chatId)) {
+          return {
+            ...el,
+            chatData: el.chatData.map((el) => {
+              if (el.createdAt === id) {
+                return target;
+              } else {
+                return el;
+              }
```

Comment:
요렇게 if문에서 많은 일을 하지 않을 땐 삼항 연산자로 슈슉 쓰는게 한눈에 더 들어올 것 같아용
```suggestion
              return el.createdAt === id ? target : el;
```